src/main/java/org/broadinstitute/hellbender/tools/exome/DiscoverCopyNumberTriStateSegments.java

```diff
@@ -0,0 +1,484 @@
+package org.broadinstitute.hellbender.tools.exome;
+
+import com.google.common.collect.Iterators;
+import org.broadinstitute.hellbender.cmdline.*;
+import org.broadinstitute.hellbender.cmdline.programgroups.CopyNumberProgramGroup;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.exome.hmm.CopyNumberTriStateHiddenMarkovModel;
+import org.broadinstitute.hellbender.tools.exome.hmm.CopyNumberTriStateHiddenMarkovModelArgumentCollection;
+import org.broadinstitute.hellbender.utils.IndexRange;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.QualityUtils;
+import org.broadinstitute.hellbender.utils.hmm.CopyNumberTriState;
+import org.broadinstitute.hellbender.utils.hmm.ForwardBackwardAlgorithm;
+import org.broadinstitute.hellbender.utils.hmm.ViterbiAlgorithm;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+
+/**
+ * Discover potential copy-number segments in input read-counts as the most likely segment sequences
+ * based on {@link CopyNumberTriStateHiddenMarkovModel} HMM model.
+ *
+ * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
+ */
+@CommandLineProgramProperties(
+        programGroup = CopyNumberProgramGroup.class,
+        summary = "Finds possible locations for rare copy number variation events in germline samples using a HMM",
+        oneLineSummary = "Discovers possible location of copy number variation"
+)
+public final class DiscoverCopyNumberTriStateSegments extends CommandLineProgram {
+
+    public static final String INPUT_ARE_ZSCORES_FULL_NAME = "inputAreZscores";
+
+    public static final String INPUT_ARE_ZSCORES_SHORT_NAME = "zscores";
+
+    /**
+     * Maximum reportable output quality score; higher quality scores will
+     * capped to this value.
+     */
+    public static final double MAX_QUAL_SCORE = 199.9999;
```

Comment:
Not really, XHMM's is 99.0 so perhaps I will change it to that but there is no actual reason for it; I think.